src/sql/workbench/browser/modelComponents/declarativeTable.component.ts

```diff
@@ -93,7 +93,7 @@ export default class DeclarativeTableComponent extends ContainerBase<any, azdata
 
 	public isCheckBox(colIdx: number): boolean {
 		let column: azdata.DeclarativeTableColumn = this.columns[colIdx];
-		return column.valueType === DeclarativeDataType.boolean;
+		return column?.valueType === DeclarativeDataType.boolean;
```

Comment:
Why would column possibly be undefined here and below?

Reply:
I was playing with the data in the import extension and one time I accidentally passed more data than columns and that caused this part of code to return error.

Reply:
Hm, it seems like we should be having the logic only fetch data for the number of columns defined instead. That'd be safer and better performing since we'd shortcut doing the unnecessary extra calls.

If you do decide to keep this the way it is though, please update the typings (e.g. `const column: azdata.DeclarativeTableColumn | undefined = this.columns[colIdx]` and ideally add a comment explaining why they might be undefined.

Reply:
@Charles-Gagnon , I have added a check in html so this should prevent any out of bound access for column data